lib/fastlane/plugin/android_emulator/actions/android_emulator_action.rb

```diff
@@ -36,11 +36,23 @@ def self.run(params)
         end
 
         UI.message("Starting emulator")
-        system("LC_NUMERIC=C; #{sdk_dir}/emulator/emulator @#{params[:name]} -port #{port} > /dev/null 2>&1 &")
-        sh("#{adb} -e wait-for-device")
 
-        until Actions.sh("#{adb} -e shell getprop dev.bootcomplete", log: false).strip == "1" do
-          sleep(5)
+        if params[:locale]          
+          system("LC_NUMERIC=C; #{sdk_dir}/emulator/emulator @#{params[:name]} -no-snapshot -change-locale #{params[:locale]} -port #{port} > /dev/null 2>&1 &")
+          sh("#{adb} -e wait-for-device")
+
+          until Actions.sh("#{adb} -e shell getprop persist.sys.locale", log: false).strip == params[:locale] && 
```

Comment:
Do we need to wait for the locale to be changed?

Reply:
Unfortunately yes. The emulator do a restart after the locale is changed.